### PR TITLE
Use dict representation for compress and decompress

### LINKED ISSUE #38: Add tests for metadata schemas across tables
Make sure we have test coverage of metadata schema round trips across all tables.

### LINKED ISSUE #39: Refactor to use the dict representation of tree sequence
Currently we're explicitly writing down the columns that we compress, leading to loss of data when we have format updates in tskit. We should use the dict representation of the tables instead, and automatically create Column objects for the data with good default compression options. We can make some special cases for particular columns if we like, but we should by default always store all the data that comes from the input tree sequence.

related to #35

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #39 
Fixes #38

### CODE REVIEW

File: tests/test_compression.py [hunk 1]
```diff
@@ -200,6 +197,45 @@ def test_mutation_parent_example(self):
         tables.mutations.add_row(site=0, node=0, parent=0, derived_state="A")
         self.verify(tables.tree_sequence())
 
+    def test_all_fields(self):
+        n = 10
+        t = 1
+        population_configurations = [
```
Comment (jeromekelleher) [inline tests/test_compression.py:203]: Easier to use ``sim_ancestry`` as this'll make individuals for you. Something like the first example [here](https://tskit.dev/msprime/docs/stable/demography.html) with ``record_migrations=True`` should do it.

File: tests/test_cli.py [hunk 2]
```diff
@@ -43,6 +42,8 @@ class TestException(Exception):
     Custom exception we can throw for testing.
     """
 
+    __test__ = False
```
Comment (jeromekelleher) [inline tests/test_cli.py:45]: What's happening here?

**Comment by jeromekelleher** [comment]
We should update the CHANGELOG with a summary of the changes. So the idea is that the tszip encoding should be identical to older versions?

Is there any way we can verify this? Maybe we should put in a few old tszipped files in repo, and verify that we recover these correctly, at least?

File: tests/test_compression.py (hunk 1 above)
Comment (benjeffery) [reply tests/test_compression.py]: Lazy copy pasting! I need get on the 1.0.0 train! Fixed.

File: tests/test_cli.py (hunk 2 above)
Comment (benjeffery) [reply tests/test_cli.py:46]: Telling pytest that isn't a test class. Added a comment.

**Comment by benjeffery** [comment]
I've added a changelog and a test of a legacy file. The idea is that old files should still work.

**Pull request merged**